src/Features/CSharp/Portable/CodeMapper/CSharpMapCodeService.NodeToMap.cs

```diff
@@ -0,0 +1,423 @@
+            {
+                matchingNode = node.DescendantNodesAndSelf()
+                                    .Where(n => IsScopedNode(n, out var scope) && scope == Scope && GetIdentifierName(n) == IdentifierName)
+                                    .FirstOrDefault();
+            }
+
+            return matchingNode is not null;
+        }
+
+        /// <summary>
+        /// Returns a string representation of the node.
+        /// </summary>
+        /// <returns>A string representation of the node.</returns>
+        public override string ToString()
+        {
+            return Node.ToFullString();
+        }
+
+        /// <summary>
+        /// Gets the identifier name of a syntax node.
+        /// </summary>
+        /// <param name="node">The syntax node to get the identifier name for.</param>
+        /// <returns>The identifier name of the syntax node.</returns>
+        private static string GetIdentifierName(SyntaxNode node)
+        {
+            return node switch
+            {
+                (MethodDeclarationSyntax method) => method.Identifier.ToString(),
+                (ClassDeclarationSyntax cl) => cl.Identifier.ToString(),
+                (InterfaceDeclarationSyntax ifc) => ifc.Identifier.ToString(),
+                (PropertyDeclarationSyntax prop) => prop.Identifier.ToString(),
+                (EnumDeclarationSyntax enu) => enu.Identifier.ToString(),
+                (StructDeclarationSyntax str) => str.Identifier.ToString(),
+                (RecordDeclarationSyntax rcr) => rcr.Identifier.ToString(),
+                (ConstructorDeclarationSyntax constructor) => BuildConstructorIdentifier(constructor),
+                (LocalFunctionStatementSyntax localFunction) => BuildLocalFunctionIdentifier(localFunction),
+                (FieldDeclarationSyntax fieldDeclaration) => BuildFieldDeclarationIdentifier(fieldDeclaration),
+                _ => node.ToString(),   // Fallback to returning the full string of the node as identifier.
+            };
+        }
+
+        private static string BuildFieldDeclarationIdentifier(FieldDeclarationSyntax fieldDeclaration)
+        {
+            var firstDeclarator = fieldDeclaration.Declaration.Variables.FirstOrDefault();
+            if (firstDeclarator is not null)
+            {
+                return firstDeclarator.Identifier.ToString();
+            }
+
+            // Fallback to returning the full string of the node as identifier.
+            return fieldDeclaration.ToString();
+        }
+
+        private static string BuildLocalFunctionIdentifier(LocalFunctionStatementSyntax localFunction)
+        {
+            var identifier = localFunction.Identifier.ToString();
+
+            // when identifier is empty, that means this
+            // local function declaration represents a Constructor
+            if (string.IsNullOrWhiteSpace(identifier))
+            {
+                return BuildConstructorIdentifier(localFunction);
```

Comment:
i don't know what this means.  "this local function declaration represents a constructor"?